service/dap/server.go

```diff
@@ -1555,6 +1562,64 @@ func (s *Server) onCancelRequest(request *dap.CancelRequest) {
 	s.sendNotYetImplementedErrorResponse(request.Request)
 }
 
+// onExceptionInfoRequest handles 'exceptionInfo' requests.
+// Capability 'supportsExceptionInfoRequest' is set 'initialize' response.
+func (s *Server) onExceptionInfoRequest(request *dap.ExceptionInfoRequest) {
+	goroutineID := request.Arguments.ThreadId
+	var body dap.ExceptionInfoResponseBody
+	// Get the goroutine and the current state.
+	g, _ := s.debugger.FindGoroutine(goroutineID)
+	state, _ := s.debugger.State( /*nowait*/ true)
```

Comment:
why ignore the errors?

Reply:
I was originally ignoring them to do a best effort approach to filling out the exception info. Though if we can't verify that this goroutine had an exception, we should return an error. Fixed.